addons/main/functions/fnc_jammerPlayerLocal.sqf

```diff
@@ -111,3 +110,36 @@ if (!isNull _drone) then {
 	};
 };
 _PP_film ppEffectCommit 0;	// commit what ever change has been made
+
+
+
+// make players connected to a drone via UAVterminal visible in the spectrum
+if (EGVAR(spectrum,UAVterminalUserVisibleInSpectrum)) then {
+	private _uav = getConnectedUAV player;
+	if (!isNull _uav) then {
+		private _uavTerminalSignalIsSet = player getVariable ["UAVTerminalSignalIsSet", false];
+		if (!_uavTerminalSignalIsSet) then {
+			// randomize frequency 
+			private _range = abs((EGVAR(spectrum,spectrumDeviceFrequencyRange)#2)#0 - (EGVAR(spectrum,spectrumDeviceFrequencyRange)#2)#1);
+			private _freq = 433.00 + (random _range);
+
+			// determine signal range
+			private _sigRange = 300;	// default value
+			{			
+				if (_x#0 == _uav) exitWith { _sigRange = _x#2 };	// use same signal range as the connected UAV
+			} forEach EGVAR(spectrum,beacons);
+
+			// add signal source
+			[QEGVAR(spectrum,addBeacon), [player, _freq, _sigRange, "drone"]] call CBA_fnc_serverEvent;
+			player setVariable ["UAVTerminalSignalIsSet", true];	// remember signal state locally
+		};
+	} else {
+		// remove signal source
+		[QEGVAR(spectrum,removeBeacon), [player]] call CBA_fnc_serverEvent;
+		player setVariable ["UAVTerminalSignalIsSet", false];	// remember signal state locally
```

Comment:
Worth only setting this variable once, and include the frequency chosen for it? Thinking in terms of right now if I connect/disconnect to the same drone with the same player, I will show up as different frequency ever time? Having it be a random frequency could make it harder to consistently track the same drone operator, using the same drone? 

(Thinking its a loop-hole your players could notice and when using drones every few minutes jump out and in again of the control to refresh frequency)

Reply:
Also a fair point. Would indeed be better if the frequency stayed the same. I will change that.

Reply:
@Crowdedlight instead of setting a fixed frequency per UAVterminal, to replace the random freq at every new connection, I chose to make it a fixed duplex offset to the drones transmit frequency (see 35aebec5b0971a3635eac877f5a6a3a5b8b4d346).

I burrowed this duplex offset logic from some reallife transmission system behaviour.

This way player could deduce that a certain pair of signals migt be a drone and its operator.

But a minor loophole/bug remains. 😕 
If one does not disconnect from one drone, but immediatly changes connection to another drone (e.g. via mouse-wheel action menu or from within UAVterminal), then the duplex offest is not yet changed to match the new drone. It instead still show the frequency used for the previously connected drone.
I try to fix this as well.

Reply:
> But a minor loophole/bug remains. 😕 If one does not disconnect from one drone, but immediatly changes connection to another drone (e.g. via mouse-wheel action menu or from within UAVterminal), then the duplex offest is not yet changed to match the new drone. It instead still show the frequency used for the previously connected drone. I try to fix this as well.

d966678505739f18998518c2ed1806e950721ec7 fixes that ☝️ 

@Crowdedlight this feedback stream is also resolved from my point of view. Do you concurr?

Reply:
Looks good. Duplex is fine using two different frequencies with a fixed offset. Makes good sense just simulating receiver pairs on both ends for full duplex. ;-)